### PR TITLE
Embed VolumeSource in v1beta3 and internal.

### LINKED ISSUE #3858: for v1beta3: embed VolumeSource
#3818 refactor health checking and Probes, and sets an example of embedding the struct holding a one-of choice.  We have another example of this in VolumeSource which does not embed.  We should change it to embed to simplify the API a bit.

### PR DESCRIPTION
**Pull request by brendandburns**
Closes #3858

@thockin @bgrant0607 @smarterclayton @markturansky

### CODE REVIEW

File: pkg/kubectl/cmd/util/helpers_test.go [hunk 1]
```diff
@@ -88,12 +88,12 @@ func TestMerge(t *testing.T) {
 				Spec: api.PodSpec{
 					Volumes: []api.Volume{
 						{
-							Name:   "v1",
-							Source: api.VolumeSource{EmptyDir: &api.EmptyDirVolumeSource{}},
+							Name:         "v1",
```
Comment (markturansky) [inline pkg/kubectl/cmd/util/helpers_test.go]: Spacing?  same below, too.

Comment (markturansky) [inline pkg/kubectl/cmd/util/helpers_test.go]: nm, i see you lined them up.

**Comment by markturansky** [comment]
LTGM.  I like this change.  It's a little less verbose overall.  I'll make the same changes to Persistent Volumes after this is merged.  Thanks for the heads up.

**Comment by bgrant0607** [comment]
LGTM. Thanks.

**Comment by thockin** [comment]
What about all the examples - I dont see any .json or .yaml files.

**Comment by thockin** [comment]
I see at least 3 files using v1b3 and volumes:

./examples/mysql-wordpress-pd/v1beta3/mysql.yaml
./examples/mysql-wordpress-pd/v1beta3/wordpress.yaml
./examples/cassandra/v1beta3/cassandra-controller.yaml

**Comment by brendandburns** [comment]
examples updated.  (fwiw, I guess we need to expand coverage of our unit test to include those files, since that unit test passes)

**Comment by markturansky** [comment]
The problem with examples_test.go is that this is valid:

```
kind: Pod
apiVersion: v1beta3
metadata:
  name: mypod
  none:
    of:
      these:
        fields:
          exist: "but the test still passes"
```

Unless validation catches a bad _value_, the tests will pass.

**Comment by thockin** [comment]
LGTM

**Pull request merged**